config/externalname.go

```diff
@@ -2233,10 +2233,10 @@ var ExternalNameConfigs = map[string]config.ExternalName{
 	//
 	// Network Firewall Policies can be imported using their ARN
 	// Example: arn:aws:network-firewall:us-west-1:123456789012:firewall-policy/example
-	"aws_networkfirewall_firewall_policy": config.TemplatedStringAsIdentifier("name", "arn:aws:network-firewall:{{ .parameters.region }}:{{ .setup.client_metadata.account_id }}:firewall-policy/{{ .external_name }}"),
+	"aws_networkfirewall_firewall_policy": config.IdentifierFromProvider,
 	// Network Firewall Rule Groups can be imported using their ARN
 	// Example: arn:aws:network-firewall:us-west-1:123456789012:stateful-rulegroup/example
-	"aws_networkfirewall_rule_group": config.TemplatedStringAsIdentifier("", "arn:aws:network-firewall:{{ .parameters.region }}:{{ .setup.client_metadata.account_id }}:stateful-rulegroup/{{ .external_name }}"),
+	"aws_networkfirewall_rule_group": config.IdentifierFromProvider,
```

Comment:
As discussed [here](https://github.com/upbound/provider-aws/issues/657#issuecomment-1507259764), we have two problems with the old configuration. We can replace the `.parameters.region` in the template expression with `{{ .setup.configuration.region }}` as suggested above but this will not be sufficient because we also need to take care of the `stateful-rulegroup` ARN component. 

However, if we extend the templating functions available for `config.TemplatedStringAsIdentifier`, we will be able to do the following:
```suggestion
	"aws_networkfirewall_rule_group": config.TemplatedStringAsIdentifier("", "arn:aws:network-firewall:{{ .setup.configuration.region }}:{{ .setup.client_metadata.account_id }}:{{ .parameters.type | ToLower }}-rulegroup/{{ .external_name }}"),
```

This will also ensure external-name normalization and backward-compatibility as discussed [here](https://github.com/upbound/provider-aws/pull/661#discussion_r1165788746). 

If we agree on this, I can open a small PR in upjet adding the `ToLower` templating function to make this possible. Also verified this configuration with imports on my local setup.

Reply:
> As discussed [here](https://github.com/upbound/provider-aws/issues/657#issuecomment-1507259764), we have two problems with the old configuration. We can replace the `.parameters.region` in the template expression with `{{ .setup.configuration.region }}` as suggested above but this will not be sufficient because we also need to take care of the `stateful-rulegroup` ARN component. 
> 
> However, if we extend the templating functions available for `config.TemplatedStringAsIdentifier`, we will be able to do the following:
> ```suggestion
> 	"aws_networkfirewall_rule_group": config.TemplatedStringAsIdentifier("", "arn:aws:network-firewall:{{ .setup.configuration.region }}:{{ .setup.client_metadata.account_id }}:{{ .parameters.type | ToLower }}-rulegroup/{{ .external_name }}"),
> ```
> 
> This will also ensure external-name normalization and backward-compatibility as discussed [here](https://github.com/upbound/provider-aws/pull/661#discussion_r1165788746). 
> 
> If we agree on this, I can open a small PR in upjet adding the `ToLower` templating function to make this possible. Also verified this configuration with imports on my local setup.

yes open the PR in upjet and provide the new version i can update this PR

Reply:
Once we consume https://github.com/upbound/upjet/pull/190, we can proceed with the suggested configuration.

Reply:
Hi @haarchri,
https://github.com/upbound/upjet/pull/190 has been merged. We can bump the upjet dependency in this PR and proceed with the configuration changes if you'd like. The template functions are available in commit `1268a48eba95ef194823f4bcc118e14a5d867a94` and later. Thank you very much.

Reply:
Start working on it